students/ZidanLuo/session09/cli_main.py

```diff
@@ -0,0 +1,148 @@
+import os
+import sys
+from donor_models import Donor
+from donor_models import Donor_Collection
+
+main_prompt = "\n".join(
+    ("Welcome to the mailroom!", "Choose an action:",
+     "1 - Input", "2 - Output", "e - Exit", ">>>  "))
+
+
+err_prompt = "\n".join(
+    ("Please provide a valid option!",
+     "1 - Input", "2 - Output", "e - Exit", ">>>  "))
+
+
+donor_in_prompt = "\n".join(
+    ("Welcome to donor database!", "Choose an action:",
+     "1 - Add a donation",  "e - Exit to main panel", ">>>  "))
+
+
+donor_out_prompt = "\n".join(
+    ("Welcome to donor database!", "Choose an action:",
+     "1 - Generate thank you letter for last donation",  "2 - Generate donation report", "3 - Save all thank you to disk", "e - Exit to main panel", ">>>  "))
+
+donorDb = Donor_Collection()
+last_donor = ""
+last_donation_amount = 0
+OUT_PATH = "thank_you_letters"
+
+def input_handler():
+    while True:
+        response = input(donor_in_prompt)
+        # create a dictionary for user's selections
+        switch_donorDB_dict = {
```

Comment:
ideally, you wold define the "switch dict" outside this function, and then could use this same function with different collections of options.